packages/store/addon/-private/identifiers/cache.ts

```diff
@@ -20,6 +20,15 @@ type ResourceIdentifierObject = import('../ts-interfaces/ember-data-json-api').R
 type ExistingResourceObject = import('../ts-interfaces/ember-data-json-api').ExistingResourceObject;
 type ConfidentDict<T> = import('../ts-interfaces/utils').ConfidentDict<T>;
 
+/**
+ * Peekable - (id && type) || lid
+ * - Identifiers are guaranteed to have a lid
+ * - ResourceIdentifierObject is Existing or New
+ *    - NewResources are guaranteed to have a lid
+ *    - ExistingResources are guaranteed to have an id AND type
+ */
+export type Peekable = RecordIdentifier | ResourceIdentifierObject;
```

Comment:
@runspired lets try to come up with a better name for this type

Reply:
Agree but for now I think this is fine and communicates the intent.